web-components/chart/chart.tsx

```diff
@@ -153,37 +155,69 @@ export class CobChart {
     this.chartID = `cob-chart-${idSuffix}`;
     let compiledSpec;
 
+    // We check to see if we are using selections.
+    // If we're using VegaLite, this is easy - we just
+    // see if the selection section of the config is not null.
+    if (this.vegaLite) {
+      this.usingSelection = this.config.selection ? true : false;
+    } else {
+      // Things get a little more complicated with Vega because
+      // signals can be more broadly used. We check to see if
+      // we're using signals, and in the array of singals we look
+      // for one we're binding to a select element.
+      if (this.config.signals) {
+        this.vegaSignal = this.config.signals.find(
+          signal => signal.bind.input === 'select'
+        );
+        this.usingSelection = this.vegaSignal ? true : false;
+      }
+    }
+
     // We check to see if the chart uses interactive selections.  If yes,
     // we manipulate the config before compiling.
-    if (typeof this.config.selection === 'undefined') {
-      // Currently, we don't support selections on charts we build with
-      // Vega (pie charts), and we only compile the spec if we're using VegaLite.
+    if (!this.usingSelection) {
+      // If we using VegaLite, we compile to VegaLite spec to Vega. It not,
+      // we leave the given config as is.
       compiledSpec = this.vegaLite
         ? VegaLite.compile(this.config).spec
         : this.config;
     } else {
       // If we are using a selection, we hold onto the field we're
       // using for it so we can populate the select options later.
-      this.selectField = this.config.selection.select.fields[0];
+      this.selectField = this.vegaLite
+        ? this.config.selection.select.fields[0]
+        : // Vega doesn't accept the name of a field as a parameter to the spec -
+          // instead we define that in the "expr" (expression) parameter accompanying
+          // the "filter" transform. So that we can populate the select options,
+          // we add it to the spec ourselves and more easily grab it here.
+          this.config.boston.selectField;
 
       // We clear the location in the config where someone could pass
       // select options as we'll build them ourselves using the data.
-      this.config.selection.select.bind.options = [];
+      if (this.vegaLite) {
+        this.config.selection.select.bind.options = [];
+      } else {
+        this.vegaSignal.bind.options = [];
+      }
 
       // After updating the config, we compile it to Vega and parse it
-      compiledSpec = VegaLite.compile(this.config).spec;
+      compiledSpec = this.vegaLite
```

Comment:
This is repeated from above. Can it be pulled out of the `if`?  

Not if it makes things harder to understand, but there's always a fair chance that someone will remember to update _one_ `compile` but not the other and not realize it's broken until they remember to test without selections or whatever.

Reply:
Pull it out of the if statement